contracts/dnsregistrar/DNSRegistrar.sol

```diff
@@ -98,11 +85,39 @@ contract DNSRegistrar {
         claim(name, proof);
     }
 
+    function proveAndClaimWithResolver(bytes memory name, DNSSEC.RRSetWithSignature[] memory input, bytes memory proof, address resolver, address addr) public {
+        proof = oracle.submitRRSets(input, proof);
```

Comment:
This and line 84 have input have `proof` variables shadowing. Maybe safer to rename the input one to `_proof` to avoid any unintended effect?

Reply:
We're intentionally replacing the call parameter with the new value.